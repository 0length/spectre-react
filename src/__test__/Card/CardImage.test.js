import React from 'react'
import { shallow } from 'enzyme'

import { CardImage } from '../..'

describe('CardImage', () => {
  it('Should render default .card-image markup', () => {
    const wrapper = shallow(<CardImage />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('card-image')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <CardImage>
        <p>hello world</p>
      </CardImage>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<CardImage className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('card-image')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<CardImage renderAs='main'>custom</CardImage>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('card-image')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<CardImage />)

    expect(wrapper).toMatchSnapshot()
  })
})
