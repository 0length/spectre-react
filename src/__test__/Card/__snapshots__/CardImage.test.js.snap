// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage Should match the snapshot 1`] = `
<div
  className="card-image"
/>
`;
